docs/tutorial.rst

```diff
@@ -0,0 +1,134 @@
+------------------------------
+
+Validation errors raised from the :func:`stereotype.Model.validate` method can report common data issues,
+such as missing required fields and broken validation rules:
+
+.. literalinclude:: ../examples/atomic_fields.py
+    :language: python
+    :lines: 33-47
+
+* The string representation of a :class:`stereotype.ValidationError` reports the first errors,
+  while the ``errors`` property can be used to identify all errors in the model structure, including nested models.
+* Not providing a value to a required field will trigger a validation error for the field. The model's attribute can
+  still be accessed but will contain the special :data:`stereotype.Missing` object.
+* Providing ``None`` to a field with a default will override the default.
+  Unless the field is ``Optional``, this also causes a validation error.
+* There are a few built-in validation rules for the specific types of :ref:`Fields`.
+  Custom validation can be implemented using validator methods.
+* Omitted required fields in the input will be missing in serialized data as well.
+
+
+Conversion errors are raised during model initialization if the input to a model wasn't a dictionary or any of
+    the fields contain data that cannot be interpreted by the corresponding field.
+
+.. literalinclude:: ../examples/atomic_fields.py
+    :language: python
+    :lines: 50-53
+
+* :class:`stereotype.ConversionError` also has the ``errors`` property with the same structure of errors,
+  but will only have one error.
+
+---------------------
+Compound value fields
+---------------------
+
+Fields can also be lists and dictionaries of other fields:
+
+.. literalinclude:: ../examples/compound_fields.py
+    :language: python
+    :lines: 3-24
+
+* Lists are created using the ``List`` type annotation and any other supported annotation, including other lists.
+* You can specify validation rules by providing an explicit :class:`stereotype.ListField`, but this is not necessary.
+  Validation rules for items may be specified by providing an explicit field instance in the ``item_field`` argument.
+* Dictionaries are created using the ``Dict`` type annotation, with an atomic key type and any value type.
+* Extra options can again be specified with an explicit :class:`stereotype.DictField`,
+  such as validation rules for the keys and values.
+* You can use the ``list`` and ``dict`` constructors as defaults if the fields are not required. ``[]`` and ``{}`` used
+  as defaults are supported as special cases for convenience and will be replaced with their respective constructor.
+
+.. literalinclude:: ../examples/compound_fields.py
+    :language: python
+    :lines: 27-40
+
+* Validation errors in compound fields will be located using nested dictionaries to the individual items.
+
+-----------------
+Nested structures
+-----------------
+
+Models can be used as model fields themselves.
+
+.. literalinclude:: ../examples/compound_fields.py
```

Comment:
You probably wanted to include `model_fields.py` here?

Reply:
Indeed, thanks